2017.md

```diff
@@ -0,0 +1,23 @@
+# Awesome Advent of Code
```

Comment:
I think it's unlikely we'll collect enough links to matter for years pre 2018, so let's limit the archive to years this repo existed.